server/connectors/cloudbees_connector/fleet_client.py

```
@@ -0,0 +1,187 @@
+
+            if service:
+                jobs = [j for j in jobs if service.lower() in (j.get("name", "") or "").lower()]
+
+            builds: List[Dict] = []
+            for job in jobs[:MAX_JOBS_PER_CONTROLLER]:
+                builds.extend(
+                    self._collect_job_builds(client, job, controller_name, controller_url.rstrip("/"), cutoff_ms)
+                )
+            return builds, None
+
+        except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)